experimental/emulated_vault/opt/emulated-vault/vault-server

```diff
@@ -0,0 +1,249 @@
+#! /usr/bin/python
+
+#
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#
+#
+
+import json
+import logging
+import sys
+import os
+import inspect
+from BaseHTTPServer import HTTPServer
+from BaseHTTPServer import BaseHTTPRequestHandler
+import ssl
+import urlparse, json
+import re, fnmatch
+
+import adaptor.fs
+
+class RequestHandler(BaseHTTPRequestHandler):
+
+    def do_GET(self):
+        try:
+            self._do_GET()
+        except:
+            logger.exception("do_GET exception")
+
+    do_HEAD = do_GET
+
+    def do_POST(self):
+        try:
+            self._do_POST()
+        except:
+            logger.exception("do_POST exception")
+
+    do_PUT = do_POST
+
+    def do_DELETE(self):
+        try:
+            self._do_DELETE()
+        except:
+            logger.exception("do_DELETE exception")
+
+    def _do_GET(self):
```

Comment:
Maybe you wanna separate out the path-specific handlers into their own methods instead of separating out all of the real logic of GET handling into a single, massive function?